src/model/src/PrismaticJoint.cpp

```diff
@@ -0,0 +1,358 @@
+/*
+ * Copyright (C) 2015 Fondazione Istituto Italiano di Tecnologia
+ * Authors: Silvio Traversaro
+ * CopyPolicy: Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+ *
+ */
+
+#include <iDynTree/Model/PrismaticJoint.h>
+
+#include <iDynTree/Core/Axis.h>
+#include <iDynTree/Core/VectorDynSize.h>
+#include <iDynTree/Core/TransformDerivative.h>
+#include <iDynTree/Model/LinkState.h>
+#include <iDynTree/Core/Wrench.h>
+#include <iDynTree/Core/Twist.h>
+
+#include <cassert>
+
+#include <cfloat>
+
+namespace iDynTree
+{
+
+PrismaticJoint::PrismaticJoint(const LinkIndex _link1, const LinkIndex _link2,
+                             const Transform& _link1_X_link2, const Axis& _translation_axis_wrt_link1):
+                             link1(_link1), link2(_link2), link1_X_link2_at_rest(_link1_X_link2),
+                             translation_axis_wrt_link1(_translation_axis_wrt_link1)
+{
+    this->setPosCoordsOffset(0);
+    this->setDOFsOffset(0);
+
+    this->resetAxisBuffers();
+    this->resetBuffers(0);
+    this->disablePosLimits();
+}
+
+PrismaticJoint::PrismaticJoint(const PrismaticJoint& other):
+                             link1(other.link1), link2(other.link2),
+                             link1_X_link2_at_rest(other.link1_X_link2_at_rest),
+                             translation_axis_wrt_link1(other.translation_axis_wrt_link1),
+                             m_hasPosLimits(other.m_hasPosLimits),
+                             m_minPos(other.m_minPos), m_maxPos(other.m_maxPos)
+{
+    this->setPosCoordsOffset(other.getPosCoordsOffset());
+    this->setDOFsOffset(other.getDOFsOffset());
+
+    this->resetAxisBuffers();
+    this->resetBuffers(0);
+}
+
+PrismaticJoint::~PrismaticJoint()
+{
+}
```

Comment:
To start using C++11 in the headers, we need to have a proper way to export the fact that the headers are using C++11 and need to be compiled with C++ >= 11, compatibly with the fact that we need to remain compatible with CMake 3.0 (see http://wiki.icub.org/wiki/YARP_Supported_Distributions). It can be done, but this are CMake modifications that are worth a separate PR.